File: day15/part1/a12.js
class Animal {
    constructor(name){
        if (this.constructor === Animal){
            throw new Error('Cannot instantiate an absctract class');
        }
        this.name = name;
    }
    speak() {
        throw new Error("Method 'speak()' must be implemented.");
    }

}

class Dog extends Animal {
    constructor(name,breed){
        super(name);
        this.breed = breed;
    }
    speak(){
        console.log(`${this.name} barks. `);
    }
}

const Animals = [
    new Dog('')
]

File: day40/a12.py
class Animal:
    def __init__(self,name,type):
        self.name = name
        self.type = type

    def speak(self):
        return f"{self.name} the smart {self.type} say hello"
    

dog = Animal("buddy","dog")
print(dog.speak())
print(dog.type)